From 47b8753f2300840893add8dcf5a97c52689a6260 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Thu, 11 Feb 2021 17:39:44 +0100
Subject: GCC: only set is_cfi for Clang builds.

There is an assert in sanitizers.gni, so if is_cfi is set,
then is_clang should be set too. So, when a GCC build is done,
it should never set is_cfi.

Added that condition explicitely to the default value of is_cfi.

Bug: 819294
Change-Id: Iea802565ea67d0ca248e0263ad009b0d6110695b
---
 build/config/sanitizers/sanitizers.gni | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/config/sanitizers/sanitizers.gni b/build/config/sanitizers/sanitizers.gni
index 39d197cf600c..3b019496a7be 100644
--- a/build/config/sanitizers/sanitizers.gni
+++ b/build/config/sanitizers/sanitizers.gni
@@ -59,7 +59,7 @@ declare_args() {
   #
   # TODO(crbug.com/1159424): Reassess the validity of the next expression.
   is_cfi =
-      is_official_build &&
+      is_official_build && is_clang &&
       (((target_os == "linux" || is_chromeos_lacros) && target_cpu == "x64") ||
        ((is_chromeos_ash || is_chromeos_lacros) && is_chromeos_device))
 
-- 
2.25.1

